### PR TITLE
[MLIR] mlir master

### PR DESCRIPTION
**Pull request by adstraw**

### CODE REVIEW

File: src/contrib/mlir/backend/pass/affine_lowerer.cpp [hunk 1]
```diff
@@ -88,9 +90,9 @@ namespace
         {                                                                                          \
         }                                                                                          \
                                                                                                    \
-        PatternMatchResult matchAndRewrite(Operation* op,                                          \
-                                           ArrayRef<Value> operands,                               \
-                                           ConversionPatternRewriter& rewriter) const override;    \
+        LogicalResult matchAndRewrite(Operation* op,                                               \
```
Comment (nmostafa) [inline src/contrib/mlir/backend/pass/affine_lowerer.cpp:93]: nit: trailing spaces

File: src/contrib/mlir/tools/ngraph-opt/ngraph_opt.cpp [hunk 2]
```diff
@@ -61,6 +61,10 @@ static llvm::cl::opt<bool>
                   llvm::cl::desc("Run the verifier after each transformation pass"),
                   llvm::cl::init(true));
 
+static llvm::cl::opt<bool> allowUnregisteredDialects("allow-unregistered-dialects",
```
Comment (nmostafa) [inline src/contrib/mlir/tools/ngraph-opt/ngraph_opt.cpp:64]: Do you need a CL for this ? I would hard-code it to false below, unless there is a reason.

Comment (adstraw) [reply src/contrib/mlir/tools/ngraph-opt/ngraph_opt.cpp:64]: Will hard-code.

File: src/contrib/mlir/backend/pass/affine_lowerer.cpp (hunk 1 above)
Comment (adstraw) [reply src/contrib/mlir/backend/pass/affine_lowerer.cpp:93]: It's a `make style` thing

**Comment by adstraw** [comment]
Converting this to a draft PR as we cannot merge MLIR PRs to master at this moment.  Using this branch as a separate MLIR master branch.

**Pull request closed without merging**